test/br/com/ianffcs/apache_http_client_ring_adapter/client_test.clj

```diff
@@ -153,3 +153,28 @@
              (catch Throwable ex
                (ex-message ex)))))))
 
+(deftest retry-execute-simple-test
+  (let [counter (atom 0)
+        max-retries 4
+        http-client (->http-client
+                      {:retry-interval 0
+                       :max-retries max-retries
+                       :retry-fn     (fn retry-fn
+                                       [^HttpResponse resp cnt _ctx]
+                                       (let [resp-status (-> resp
+                                                             .getStatusLine
+                                                             .getStatusCode)]
+                                         (if (and (= 404 resp-status)
+                                                  (> max-retries cnt))
+                                           (do (swap! counter inc)
+                                               true)
+                                           false)))
+                       :ring-handler (fn [_req]
```

Comment:
Add a counter inside ring-handler. we need to make sure that the ring-handler was called, not the "retry fn"